bayesbridge/design_matrix/abstract_matrix.py

```diff
@@ -63,12 +70,16 @@ def toarray(self):
         """ Returns a 2-dimensional numpy array. """
         pass
 
+    @staticmethod
+    def is_cupy_sparse(X):
+        return isinstance(X, cpx.scipy.sparse.spmatrix) if cp and cpx else False
+
     @staticmethod
     def remove_intercept_indicator(X):
-        if sp.sparse.issparse(X):
-            col_variance = np.squeeze(np.array(
-                X.power(2).mean(axis=0) - np.power(X.mean(axis=0), 2)
-            ))
+        squeeze, array, power = (cp.squeeze, cp.array, cp.power) if \
```

Comment:
at some point it might be easier to doing something like `linalg_pkg = cp if self.use_cupy else np`...

Reply:
`linalg_pkg = cp if self.use_cupy else np` is a great idea. (The name `linalg_pkg` might be confusing, though; it sounds like a type of BLAS. Off top of my head, `array_lib` comes to my mind, though still not great.)